### PR TITLE
Added pagination in policy groups list

### PR DESCRIPTION
**Pull request by chaitali-mane**
Signed-off-by: chaitali-mane <cmane@progress.com>

### :nut_and_bolt: Description: What code changed, and why?
Need to add pagination functionality for Policy groups.

### :chains: Related Resources
https://github.com/chef/automate/issues/5390

### :+1: Definition of Done
Added pagination UI and functionality for Policy groups list.

### :athletic_shoe: How to Build and Test the Change
```
STEP 1
inside the hab studio

[default:/src:0]# build components/automate-ui-devproxy/
[default:/src:0]# start_automate_ui_background
[default:/src:0]# start_all_services

STEP 2
open new window
go to automate UI path

$ cd components/automate-ui
and run the command 

npm run serve:hab
```
Steps for Policy Groups tab:
Go To Infrastructure -> Chef Servers -> click server name-> click on org name -> Policy Groups tab
Where you can see the pagination for a list of groups.

For cypress Build
https://github.com/chef/automate/tree/master/e2e

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

**Review by vinay033 — APPROVED** [review]

File: components/automate-ui/src/app/modules/infra-proxy/policy-groups-list/policy-groups-list.component.html [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-  <chef-thead *ngIf="policyGroups.length">
+  <chef-thead *ngIf="policyFiles.length">
     <chef-tr class="no_border_tr">
       <chef-th class="no_border_th">Policy Group</chef-th>
       <chef-th class="no_border_th">Number of Policyfiles</chef-th>
     </chef-tr>
   </chef-thead>
   <chef-tbody>
-    <chef-tr *ngFor="let policyGroup of policyGroups">
+    <chef-tr *ngFor="let policyGroup of policyFiles">
```
Comment (himanshi-chhabra) [inline components/automate-ui/src/app/modules/infra-proxy/policy-groups-list/policy-groups-list.component.html:9]: Names are not matching we have policygroups here why change?

File: components/automate-ui/src/app/modules/infra-proxy/policy-groups/policy-groups.component.html [hunk 2]
```diff
@@ -18,8 +18,29 @@
     <chef-loading-spinner class="full-screen-spinner" *ngIf="searching" size="50" fixed></chef-loading-spinner>
     <div id="policy-group-table-container" *ngIf="policyFiles.length" data-cy="policy-group-table-container">
       <app-policy-groups-list
-        [policyFiles]= "groupList">
+        *ngIf="!searchFlag"
+        [policyFiles]= "policyGroups"
```
Comment (himanshi-chhabra) [inline components/automate-ui/src/app/modules/infra-proxy/policy-groups/policy-groups.component.html:22]: name should be with group no policy files we are showing so it should be group on left hand side

File: components/automate-ui/src/app/modules/infra-proxy/policy-groups/policy-groups.component.html [hunk 3]
```diff
@@ -18,8 +18,29 @@
...
+        [pageOfItems]= "pageOfItems">
+      </app-policy-groups-list>
+      <app-policy-groups-list
+        *ngIf="searchFlag"
+        [policyFiles]= "groupList"
```
Comment (himanshi-chhabra) [inline components/automate-ui/src/app/modules/infra-proxy/policy-groups/policy-groups.component.html:27]: same as above

File: components/automate-ui/src/app/modules/infra-proxy/policy-groups/policy-groups.component.ts [hunk 4]
```diff
@@ -35,8 +35,10 @@ export class PolicyGroupsComponent implements OnInit, OnDestroy {
   public authFailure = false;
   public searching = false;
   public searchValue = '';
-  public searchArr;
   public groupList = [];
+  public pageOfItems: Array<any>;
+  public searchFlag: boolean;
+  public policyGroups;
```
Comment (himanshi-chhabra) [inline components/automate-ui/src/app/modules/infra-proxy/policy-groups/policy-groups.component.ts:41]: Add type for this variable

**Review by himanshi-chhabra — CHANGES_REQUESTED** [review]

File: components/automate-ui/src/app/modules/infra-proxy/policy-groups/policy-groups.component.ts (hunk 4 above)
Comment (chaitali-mane) [reply components/automate-ui/src/app/modules/infra-proxy/policy-groups/policy-groups.component.ts]: added type.

File: components/automate-ui/src/app/modules/infra-proxy/policy-groups/policy-groups.component.html (hunk 2 above)
Comment (chaitali-mane) [reply components/automate-ui/src/app/modules/infra-proxy/policy-groups/policy-groups.component.html]: updated changes

File: components/automate-ui/src/app/modules/infra-proxy/policy-groups/policy-groups.component.html (hunk 3 above)
Comment (chaitali-mane) [reply components/automate-ui/src/app/modules/infra-proxy/policy-groups/policy-groups.component.html]: updated changes

File: components/automate-ui/src/app/modules/infra-proxy/policy-groups-list/policy-groups-list.component.html (hunk 1 above)
Comment (chaitali-mane) [reply components/automate-ui/src/app/modules/infra-proxy/policy-groups-list/policy-groups-list.component.html]: updated changes

**Review by himanshi-chhabra — APPROVED** [review]
LGTM Thanks @chaitali-mane :+1:

**Pull request merged**